fedimint-core/src/task.rs

```diff
@@ -31,23 +32,41 @@ type JoinError = anyhow::Error;
 #[error("deadline has elapsed")]
 pub struct Elapsed;
 
-#[derive(Debug, Default)]
+#[derive(Debug)]
 struct TaskGroupInner {
     /// Was the shutdown requested, either externally or due to any task
     /// failure?
     is_shutting_down: AtomicBool,
```

Comment:
`is_shutting_down` is redundant now. We can use `*on_shutdown_tx.borrow()` instead.